source/GameInterface/Services/BesiegerCamps/SiegeStrategyRegistry.cs

```diff
@@ -0,0 +1,26 @@
+﻿using GameInterface.Services.Registry;
+using System;
+using TaleWorlds.CampaignSystem.Siege;
+
+namespace GameInterface.Services.BesiegerCamps;
+
+// Not sure where this goes..
+internal class SiegeStrategyRegistry : RegistryBase<SiegeStrategy>
+{
+    public SiegeStrategyRegistry(IRegistryCollection collection) : base(collection)
+    {
+    }
+
+    public override void RegisterAll()
+    {
+        foreach (var strategy in SiegeStrategy.All)
+        {
+            RegisterNewObject(strategy, out var _);
+        }
+    }
+
+    protected override string GetNewId(SiegeStrategy obj)
+    {
+        return Guid.NewGuid().ToString();
```

Comment:
Here you could instead use `Campaign.Current.CampaignObjectManager.FindNextUniqueStringId<SiegeStrategy>(SiegeStrategyIdPrefix)`

Reply:
Which would look something like this " public static readonly string HeroStringIdPrefix = "CoopHero"; "

Reply:
At a second glance, a registry for SiegeEvent might not even be needed?, it's one of those "constant" objects that always exist for all clients and even has predefined StringId, all tests run fine with this class commented out

Reply:
If new SiegeStrategies are never added, nor change data then yes a registry is not needed